provider/win_driver_utils.py

```diff
@@ -57,7 +57,11 @@ def uninstall_driver(session, test, devcon_path, driver_name,
     if status:
         test.error("Not found devcon.exe, details: %s" % output)
     LOG_JOB.info("Uninstalling previous installed driver")
-    for inf_name in _pnpdrv_info(session, device_name, ["InfName"]):
+    # find the inf name and remove the repeated one
+    inf_list_all = _pnpdrv_info(session, device_name, ["InfName"])
+    inf_list = list(set(inf_list_all))
+    inf_list.sort(key=inf_list_all.index)
+    for inf_name in inf_list:
```

Comment:
Hi @xiagao Could you explain why we need to sort the inf_list.Thanks!

Reply:
Just to keep it in original order, it's ok to remove this line, but I would like to keep it.

Reply:
If we remove duplicates, inf_list should have only one item. Because the same device will use the same inf file, right?

Reply:
yes, it is.

Reply:
Suggest to convert `inf_list_all` to string directly and delete the "for" struct.

```
inf_list_all = _pnpdrv_info(session, device_name, ["InfName"])
inf_name = string(set(inf_list_all))
uninst_store_cmd = "pnputil /f /d %s" % inf_name
status, output = session.cmd_status_output(uninst_store_cmd, INSTALL_TIMEOUT)
```

Reply:
Yes, this function is only for one type of driver, there is no need to use for loop.
Thanks.